base/reduce.jl

```diff
@@ -396,7 +396,11 @@ end
 
 
 ## prod
+"""
+    prod(f, iter)
```

Comment:
It looks fine, the only issue is that it says `f::Callable`, which is a little confusing given that `Callable` isn't exported... What do you think should be done here? Is that okay or should I try something else?